dbs-device: add as_any for DeviceIo trait

wllenyj opened the pull request:
Sometimes we need to downcast out the specific type.

wllenyj committed:
dbs-device: add as_any for DeviceIo trait

Sometimes we need to downcast out the specific type.

Signed-off-by: wllenyj <wllenyj@linux.alibaba.com>

jiangliu commented on crates/dbs-device/src/lib.rs:
```diff
@@ -2,8 +2,6 @@
 // Copyright © 2019 Intel Corporation. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-#![deny(missing_docs)]
```
Why removing the `#![deny(missing_docs)]`?

wllenyj replied:
Sorry, I mistook it for `allow(missing_docs)`.
removed this change.

jiangliu commented on crates/dbs-virtio-devices/src/mmio/mmio_v2.rs:
```diff
@@ -339,8 +339,8 @@ where
-    Q: QueueStateT + Send + Clone,
-    R: GuestMemoryRegion + Send + Sync,
+    Q: QueueStateT + Send + Clone + 'static,
+    R: GuestMemoryRegion + Send + Sync + 'static,
```
`'static + QueueStateT + Send + Clone` to keep same style of above line?

wllenyj replied:
fixed

wllenyj committed:
dbs-virtio-devices: Fix unit test failure.

```
---- vsock::backend::inner::tests::test_inner_stream_timeout stdout ----
---- vsock::backend::inner::tests::test_inner_stream_timeout stderr ----
thread '<unnamed>' panicked at 'assertion failed:
(150..200).contains(&end_time1)',
crates/dbs-virtio-devices/src/vsock/backend/inner.rs:889:17
```
In this nested virtualized test environment, the execution is very slow,
and we should be more lenient in the time-dependent case.

Signed-off-by: wllenyj <wllenyj@linux.alibaba.com>
